### PR TITLE
Week 02 coding tasks [Linked list]

### PR DESCRIPTION
**Pull request by gitcseme**

### CODE REVIEW

File: week02/Add_Two_Numbers.cpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        ListNode* dummyHead = new ListNode();
+        ListNode* current = dummyHead;
+        int carry = 0;
+
+        while (l1 || l2) {
+            int x = l1 != nullptr ? l1 -> val : 0;
+            int y = l2 != nullptr ? l2 -> val : 0;
+
+            int sum = x + y + carry;
+            current -> next = new ListNode(sum % 10);
```
Comment (Sadman007) [inline week02/Add_Two_Numbers.cpp:16]: As you're creating new nodes with `new`, the space complexity can never be constant. :)

File: week02/Delete_Node_in_a_Linked_List.cpp [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// TC: O(1)
+// MC: O(1)
+
+class Solution {
+public:
+    void deleteNode(ListNode* node) {
+        node -> val = node -> next -> val;
+        node -> next = node -> next -> next;
```
Comment (Sadman007) [inline week02/Delete_Node_in_a_Linked_List.cpp:8]: You need to "delete" the actual node. This is not correct.

**Review by Sadman007 — APPROVED** [review]
Score: 8.5/10

File: week02/Add_Two_Numbers.cpp (hunk 1 above)
Comment (gitcseme) [reply week02/Add_Two_Numbers.cpp:16]: Thanks for pointing it out vai. I missed the memory here. I am creating N nodes inside the loop, so it is O(N) MC.

**Pull request merged**